internal/services/containers/container_registry_cache_rule_resource.go

```diff
@@ -0,0 +1,216 @@
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			cacheRulesClient := metadata.Client.Containers.ContainerRegistryClient_v2023_07_01.CacheRules
+			ctx, cancel := timeouts.ForRead(metadata.Client.StopContext, metadata.ResourceData)
+			defer cancel()
+
+			id, err := cacherules.ParseCacheRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := cacheRulesClient.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					log.Printf("[DEBUG] Container Registry Cache Rule %s was not found.", *id)
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving Container Registry Cache Rule %s: %+v", *id, err)
+			}
+
+			metadata.ResourceData.Set("name", id.CacheRuleName)
+			metadata.ResourceData.Set("registry", id.RegistryName)
+
+			if model := resp.Model; model != nil {
+				if properties := model.Properties; properties != nil {
+					metadata.ResourceData.Set("source_repo", properties.SourceRepository)
+					metadata.ResourceData.Set("target_repo", properties.TargetRepository)
+				}
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ContainerRegistryCacheRule) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			cacheRulesClient := metadata.Client.Containers.ContainerRegistryClient_v2023_07_01.CacheRules
+			ctx, cancel := timeouts.ForRead(metadata.Client.StopContext, metadata.ResourceData)
+
+			defer cancel()
+			log.Printf("[INFO] preparing arguments for Container Registry Cache Rule update.")
+
+			id, err := cacherules.ParseCacheRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			// TODO: You can only update the credential set. To be implemented
+			parameters := cacherules.CacheRuleUpdateParameters{
+				Properties: &cacherules.CacheRuleUpdateProperties{},
+			}
+
+			if err := cacheRulesClient.UpdateThenPoll(ctx, *id, parameters); err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (ContainerRegistryCacheRule) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			cacheRulesClient := metadata.Client.Containers.ContainerRegistryClient_v2023_07_01.CacheRules
+			ctx, cancel := timeouts.ForDelete(metadata.Client.StopContext, metadata.ResourceData)
+			defer cancel()
+
+			id, err := cacherules.ParseCacheRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := cacheRulesClient.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting Container Registry Cache Rule %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (ContainerRegistryCacheRule) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.ContainerRegistryCacheRuleID
```

Comment:
I believe you can replace the `validate` package with the content of `cacherules` in the new SDK, like this:
```suggestion
	return cacherules.ValidateCacheRuleID
```